docs/SPEC.md

```diff
@@ -1134,18 +1194,30 @@ The _null_ value can be of any data type.
 ### Transformations
 
 Transformations define a change to a stream.
-Transformations may consume an input stream and always produce a new output stream.
+Transformations must consume an input stream and always produce a new output stream.
 
-Most transformations output one table for every table they receive from the input stream.
-Transformations that modify the group keys or values will need to regroup the tables in the output stream.
-A transformation produces side effects when it is constructed from a function that produces side effects.
+Transformations always operate on the data channel within the stream.
+Transformations do not operate on the meta channel within the stream, however they may add new tables to it.
+The meta channel may contain arbitrary data about the processing of the data channel.
+For example an transformation may add a table to the meta channel containing error that were encountered.
+
+Most transformations output one table for every table they receive from the input data channel.
```

Comment:
let's give these names.   `simple transformations` and `channel transformations`  ?  `grouping transformations` ?

Reply:
Is it really useful to say this? It will be clear from examples. Or, maybe, this is the moment to provide an example for clarification after this great introduction to channels and streams.